### PR TITLE
logindex: stand the backfill down at 2 pending blocks, not 8

### PR DESCRIPTION
**Pull request by biafra23**
## Why

`BACKFILL_HEAD_GAP_TOLERANCE` was a time budget in disguise: eight blocks were justified as "well inside the 10 minutes a Bee node gives its RPC before shutting down". That measures the wrong thing. Bee's shutdown clock is not what a lagging edge costs — a head-reaching `eth_getLogs` is REFUSED for as long as the top lags **at all**, so every request in that window fails while the walk, still inside its tolerance, keeps spending the shared snap pool.

A Bee full node on zbox died exactly there (2026-09-17): ready at 00:23, 39,225 logs served through Myotis, chequebook deployed via `eth_sendRawTransaction`, then 134 consecutive `requested range is not indexed yet` refusals from 00:38 and `postage syncing stalled` with self-shutdown at 01:01:48. Across those 23 minutes coverage advanced 223 blocks while the chain advanced ~282.

## What changes

- **`BACKFILL_HEAD_GAP_TOLERANCE` 8 → 2.** Two pending blocks *are* the steady state of a 5 s chain sampled on the 6 s tick; three mean the appender is losing ground. The not-yielding guards (bridge span, stalled finality) are untouched — this only moves the point at which head-follow takes priority.
- **`BACKFILL_YIELD_MAX_TICKS` 50 → 10**, from the review. The two constants are coupled: together they set the walk's worst-case duty cycle in whatever band the trigger newly covers. At tolerance 8 that band was effectively unreachable; at 2, `pending 3..=8` is reachable by a **healthy but perpetually trailing** head-follow (the tail's per-tick budget truncates candidate-dense ranges). A 50-tick floor would have throttled the walk ~50x there for no gain, since a trailing edge refuses head-reaching queries either way. Ten ticks leave head-follow 90% of ticks.
- Docs: the coupling is now recorded on both constants, and the caller's rationale no longer argues from Bee's shutdown clock, which the new text explicitly disowns.

## Verification

- `cargo test -p myotis-net --lib -- el::reader::tests`: 37 green. The boundary cases in `backfill_yields_only_while_head_follow_can_close_the_gap` are moved to the new threshold. The review caught that my first draft put `finalized` **above** the head there, which made the `TAIL_MAX_ABOVE_FINALITY` guard pass vacuously — both cases now keep finality at or below the head.
- Live on zbox: engine rebuilt with both constants, daemon restarted (SYNCED, index 3 blocks behind the head), Bee restarted and `ready` in 90 s, 4 blocks behind the tip, zero `range not indexed` warnings so far. A watch is running.

## Honest framing

This is a mitigation, not a proven root-cause fix. The incident data show head-follow was also slow on its own in that window (223 blocks against the chain's 282), which points at a sluggish peer pool; the walk made it worse but was not solely responsible. Only a longer run will show whether this is enough.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01WTPh82Wm2SnpcQvsHsghHo

### CODE REVIEW

File: rust/myotis-net/src/el/reader.rs [hunk 1]
```diff
@@ -2933,8 +2956,9 @@ impl ElReader {
         // the appender abandons its whole tick on a single failed receipts read
         // — so the walk was buying its progress out of the head's. Nothing waits
         // on the walk, while a head-reaching `eth_getLogs` is REFUSED for as
-        // long as the top lags, and behind that refusal sits Bee's 10-minute
-        // shutdown clock.
+        // long as the top lags AT ALL — which is the cost that matters, not any
+        // consumer's shutdown timer (see `BACKFILL_HEAD_GAP_TOLERANCE`, which
+        // was once sized against Bee's 10-minute clock and is no longer).
```
[the remark this answers is not in the document]
Comment (biafra23) [reply rust/myotis-net/src/el/reader.rs:2961]: **Accepted.** Fixed in 97913be9: the yield branch now says `~10x optimistic` and names `BACKFILL_YIELD_MAX_TICKS`, so the next retune finds it from the constant. You are right that this PR of all PRs should not leave a stale number behind — it was 20 lines outside the diff and I missed it.

File: rust/myotis-net/src/el/reader.rs [hunk 2]
```diff
@@ -6202,10 +6226,11 @@ mod tests {
...
-        // The tolerance itself, counted inclusively over edge..=head: eight
-        // pending blocks stay with the walk, nine hand the tick to head-follow.
-        assert!(!backfill_should_yield(1_000, 1_007, 1_004, 0));
-        assert!(backfill_should_yield(1_000, 1_008, 1_004, 0));
+        // The tolerance itself, counted inclusively over edge..=head: two
+        // pending blocks stay with the walk (the steady state of a 5 s chain on
+        // a 6 s tick), three hand the tick to head-follow.
+        assert!(!backfill_should_yield(1_000, 1_001, 1_000, 0));
+        assert!(backfill_should_yield(1_000, 1_002, 1_001, 0));
         // Coverage past the head (edge = head + 1) is zero pending, not one.
```
[the remark this answers is not in the document]
Comment (biafra23) [reply rust/myotis-net/src/el/reader.rs:6239]: **All three accepted**, in 97913be9.

1. The opener now pins `pending = 1` (`1_000, 1_000, 1_000`) instead of duplicating the at-tolerance call, so the steady-state example and the boundary no longer pin the identical inputs.
2. The pair holds `finalized` fixed at `1_000` and moves only `head` (`1_001` stay, `1_002` yield), which makes it a single-variable boundary again, matching the neighbouring TAIL_MAX and BRIDGE_MAX pairs.
3. Added `assert!(backfill_should_yield(1_000, 1_200, 1_150, BACKFILL_YIELD_MAX_TICKS - 1))`. Cheap insurance as you say, and worth more now: the floor is the only thing between a permanently trailing head and a permanently silenced walk, and it just dropped 5x.
